src/_pytest/fixtures.py

```diff
@@ -1535,9 +1535,10 @@ def _getautousenames(self, node: nodes.Node) -> Iterator[str]:
 
     def _getusefixturesnames(self, node: nodes.Item) -> Iterator[str]:
         """Return the names of usefixtures fixtures applicable to node."""
-        for mark in node.iter_markers(name="usefixtures"):
+        for marker_node, mark in node.iter_markers_with_node(name="usefixtures"):
             if not mark.args:
-                warnings.warn(f"Warning: empty usefixtures in {node.name}.")
+                location = getattr(marker_node, "location", "unknown location")
```

Comment:
as per https://github.com/pytest-dev/pytest/blob/main/src/_pytest/nodes.py#L271 we tell the location of the node when warning, so why not just use `marker_node.warn`

Reply:
I used `marker_node.location`, but sometimes I got an attribute error: node has no attribute 'location.' I can't seem to replicate the error. Would you prefer `marker_node.warn`?

Reply:
Yes, that Will Auto fill the warning file and line